src/version_file.rs

```diff
@@ -49,20 +56,17 @@ impl<'a> VersionFile<'a> {
 }
 
 #[cfg(not(test))]
-#[cfg(any(target_os = "macos", target_os = "linux"))]
-fn home_path() -> PathBuf {
-    PathBuf::from(env!("HOME"))
-}
-
-#[cfg(not(test))]
-#[cfg(target_os = "windows")]
-fn home_path() -> PathBuf {
-    PathBuf::from(env!("USERPROFILE")) // Or use $HOMEPATH?
+fn cache_path() -> Result<PathBuf, Error> {
+    let project_dir = directories::ProjectDirs::from("", "", "update-informer-rs")
+        .map_or(Err("Unable to find cache directory"), Ok)?;
+    let directory = project_dir.cache_dir().to_path_buf();
```

Comment:
What do you think about using the [dirs](https://github.com/dirs-dev/dirs-rs) crate instead of this?
I think it has more simple API:
```
dirs::cache_dir();
```

Reply:
That gives you the general cache directory (e.g. `~/.cache` on Linux), but not the application-specific cache directory (`~/.cache/yourapp/` on Linux), which is what you probably want. `directories` is a higher level wrapper on top of that adds that logic in OS-specific way, so seems worth using otherwise you'd have to recreate the same lgoic.